sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRegistryRecordedTestBase.cs

```diff
@@ -42,12 +43,36 @@ public ContainerRegistryClient CreateClient(bool anonymousAccess = false, string
                 ));
         }
 
-        public async Task ImportImageAsync(string registry, string repository, string tag)
+        protected string GetPlatformSuffix()
         {
-            await ImportImageAsync(registry, repository, new List<string>() { tag });
+            var os = FormatIdentifier(RuntimeInformation.OSDescription);
+            var dotnetVersion = FormatIdentifier(RuntimeInformation.FrameworkDescription);
+            return $"-{os}-{dotnetVersion}";
         }
 
-        public async Task ImportImageAsync(string registry, string repository, List<string> tags)
+        private string FormatIdentifier(string value)
+        {
+            List<string> invalidCharacters = new List<string> { " ", ".", "#", "~", ":", ";", "/", "\\" };
+            foreach (var invalid in invalidCharacters)
+            {
+                value = value.Replace(invalid, string.Empty);
+            }
+
+            int maxLength = 25;
+            if (value.Length > maxLength)
+            {
+                value = value.Substring(0, maxLength);
```

Comment:
nit: If a SubString ends up being needed, it'd be better to do that before replacing values.